<!doctype html>
<html lang="en">
    <head>
        <script type="text/x-mathjax-config" src="../scripts/mathjax_conf.js"></script>
        <script type="text/javascript" async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/latest.js?config=TeX-AMS-MML_HTMLorMML" async>
        </script>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <!-- <title>Hakyll Blog - Formal models of natural processes should be the ultimate goal of quantitative descriptions in science </title> -->
        <link rel="stylesheet" href="../css/default.css" />
        <link rel="shortcut icon" type="image/png" href="images/favicon.png">
    </head>
    <header>
        <!-- <div class="logo">
            <a href="/">Hakyll Blog</a>
        </div> -->
      <nav>
        <ul>
          <li><a href="../">main</a></li>
  	      <li><a href="../sciphi.html">sci-phi</a></li>
          <li><a href="../publications.html">publications</a></li>
          <li><a href="../archive.html">blog</a></li>
        </ul>
      </nav>
    </header>
    <body>
        <main role="main">
            <h1>Formal models of natural processes should be the ultimate goal of quantitative descriptions in science </h1>
            <article>
    <section class="header">
    </section>
    <section>
        <p><img src="../images/experiment_span.png" class="textwidth"></p>
<p>Like it or not, the machine learning hype, well, hyped up the interest in statistics and even more basic math-related subjects like linear algebra, differential calculus, and stuff that would made most my then-little friends in school cringe and decide to become lawyers right there and then, probably to the detriment of the world at large. Somewhat like its wiser and less maligned cousin, statistics, machine learning is a broad class of methods that aim at describing different kinds of data ranging from scientific experiments to add-clicking on whatever shitty social media site is (directly or indirectly) selling your data today. That is all fine and well, but as a scientist I am interested not in describing the data per se but in associating the observed data to the processes generating them — there is a (subtle apparently) difference there that often goes underappreciated.</p>
<p>The limitations of many data-driven approaches have been criticized in different ways. Judea Pearl’s <a href="http://bayes.cs.ucla.edu/">research</a> (as well as his recent popular exhibit <a href="http://bayes.cs.ucla.edu/WHY/">Book of Why</a> and the technical text <a href="https://www.cambridge.org/us/academic/subjects/philosophy/philosophy-science/causality?format=HB&amp;isbn=9780521895606">Causality</a> pointed out the lack of the “arrow of time” and causal links in statistical analyses. More generally, the majority of approaches probably lack mechanistic insight, and law-like knowledge that is composable into a larger scale understanding. Being model-based on its own does not necessarily fix these issues. Linear models, for instance, can be very useful to describe simple trends, but the parameters most likely have no explicit meaning. When fitted to a bacterial growth curve, for instance, the line intercept can be interpreted as the initial number of bacteria, but the slope really means nothing; in contrast the exponential growth parameter can be interpreted as the rate at which they double (in log scale these can be equivalent). Linear models (or their generalized, but still essentially linear versions) are the core of methods to analyze data like gene expression (Love et al. 2014, Ritchie et al. 2015), but each gene is treated as a linear model independent from all other genes. There are few things I can think of that are more non-linear and coupled than genetic networks, and yet that is the state-of-the-art of analysis of the transcription of DNA into RNA, translation to protein and other modified products. The point applies a lot more broadly to trend-capturing models that on the other hand do not describe actual natural processes. In contrast to that are, for instance, dynamic models of well-studied gene networks like those of circadian clocks (Goldbeter 2002). Those rely on detailed knowledge of interaction of genes and gene products, and aim at explaining the observed dynamic patterns. Ideally, they should also allow identifying unknown components or less-detailed roles of the known ones, and even missing components and their roles. Nevertheless, it is unlikely that this approach can be scaled up from a couple of dozen to include thousands of genes and validated directly against genome-scale data sets with the tools available presently. Moving from particular observations to general laws is a challenging task at the core of science and a recurring subject in its philosophy; it is not straightforward, so different scientists will have different ideas for how to go about it. I’ll try to loosely illustrate some of the approaches, particularly in biology as in the above examples of genomic data analysis. They could range from a very abstract approach to find a very general pattern, to a very pragmatic approach to get specific answers. The first extreme could result in a “soft” answer, while the second is more likely to result in a pipeline of procedural or algorithmic steps that could be informal and difficult to integrate into a modular framework. In my opinion there is unfortunately no “happy middle”, but something is needed that accounts for smaller scale patterns as well as integrating modularly into general laws. If that’s not asking too much. I don’t have a simple answer (and probably not a full complex one either) for a framework that permeates all the scales. I think some of it can be achieved by addressing the limitations of mathematical modeling and statistical inference as we try to model increasingly large systems and infer their parameters. But alternatively, if may be necessary to use a different formal framework, like the mathematics of sets and category theory, which may not model quantitative relationships as strictly as our usual equations can, but may allow formal relationships between different modules and experimental designs. That is a question that is unlikely to be settled any time soon and is fertile ground for the philosophy of science. Whatever it is I think there is only hope of real progress if we can make the relationship between everything that is studied (within and between systems) formal in the mathematical sense so that it cohesively builds up towards something.</p>
<ol type="1">
<li>Judea Pearl, Dana Mackenzie. The Book of Why. <strong>2018</strong> Basic Books, New York.</li>
<li>Judea Pearl. Causality. <strong>2009</strong> Cambridge University Press.</li>
<li>Love, Huber, Anders. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biology <strong>2014</strong>; 15:550</li>
<li>Ritchie, Phipson, Wu et al. limma powers differential expression analyses for RNA-sequencing and microarray studies. Nucleic Acids Research <strong>2015</strong>; 43:e47,</li>
<li>Goldbeter. Computational approaches to cellular rhythms. Nature <strong>2002</strong>; 420:238–245</li>
</ol>

        <code>
        
            -- caetano,
         November 26, 2018
        </code>
    </section>
</article>

        </main>

        <footer>
          <ul>
            <li><a href="https://github.com/caesoma">
              <img src="../images/gh.png" alt="github" style="width:42px;height:42px;border:0;">
            </a></li>
            <li><a href="https://orcid.org/0000-0002-0271-2576">
              <img src="../images/id.png" alt="orcid" style="width:42px;height:42px;border:0;">
            </a></li>
            <li><a href="https://www.linkedin.com/in/caetanosoutomaior/">
              <img src="../images/li.png" alt="linkedin" style="width:42px;height:42px;border:0;">
            </a></li>
            <li><a href="https://mastodon.social/@caesoma">
              <img src="../images/mt.png" alt="mastodon" style="width:42px;height:42px;border:0;">
            </a></li>
            <li><a href="https://medium.com/@caesoma">
              <img src="../images/md.png" alt="medium" style="width:42px;height:42px;border:0;">
            </a></li>
            <!-- <li><a href="https://www.researchgate.net/profile/Caetano_Souto-Maior">
              <img src="/images/rg.png" alt="researchgate" style="width:42px;height:42px;border:0;">
            </a></li>
            <li><a href="https://nih.academia.edu/CaetanoSoutoMaior">
              <img src="/images/ac.png" alt="academia.edu" style="width:42px;height:42px;border:0;">
            </a></li> -->
        </ul>

          <p>
          <i>This page is licensed under <a href="https://www.gnu.org/licenses/gpl-3.0.en.html">GNU General Public License version 3</a>, which means its content can be reused and distributed, as long as it is also made available in that way, I think. You'd really have to read <a href="https://caesoma.github.io/LICENSE.md">the GPL license</a> and find out more <a href="https://choosealicense.com/">how this stuff works</a>.</i>
          </p>
          <p>
          <i>created with</i> <a href="http://jaspervdj.be/hakyll">Hakyll</a>
          </p>
          </footer>
    </body>
</html>
